v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubStreamLT.java

```diff
@@ -0,0 +1,171 @@
+import com.google.cloud.teleport.it.gcp.TemplateLoadTestBase;
+import com.google.cloud.teleport.it.gcp.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.gcp.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.gcp.datagenerator.DataGenerator;
+import com.google.cloud.teleport.it.gcp.pubsub.DefaultPubsubResourceManager;
+import com.google.cloud.teleport.it.gcp.pubsub.PubsubResourceManager;
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.common.base.MoreObjects;
+import com.google.common.io.Resources;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.OrderWith;
+import org.junit.runner.RunWith;
+import org.junit.runner.manipulation.Alphanumeric;
+import org.junit.runners.JUnit4;
+
+/** Performance tests for {@link TextToPubsubStreamLT GCS Text to PubSub} template. */
+@Category(TemplateLoadTest.class)
+@TemplateLoadTest(TextToPubsubStream.class)
+@RunWith(JUnit4.class)
+@OrderWith(Alphanumeric.class)
+public class TextToPubsubStreamLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(),
+          "gs://dataflow-templates/latest/Stream_GCS_Text_to_Cloud_PubSub");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR =
+      TextToPubsubStreamLT.class.getSimpleName().toLowerCase();
+  private static final String INPUT_PATH = "TextToPubsubStream/input.txt";
+  private static final String INPUT_PCOLLECTION =
+      "Read Text Data/Via ReadFiles/Read all via FileBasedSource/Read ranges.out0";
+
+  private static PubsubResourceManager pubsubResourceManager;
+  private static ArtifactClient gcsClient;
+  private static String generatorSchemaPath;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    // upload schema files and save path
+    generatorSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            gcsClient
+                .uploadArtifact(
+                    "schema.json",
+                    Resources.getResource("TextToPubsubStream/schema.json").getPath())
+                .name());
+
+    // upload input.txt
+    gcsClient.uploadArtifact("input.txt", Resources.getResource(INPUT_PATH).getPath());
```

Comment:
how do you use input.txt file? Does it make sense to upload it?

Reply:
I used the standard scheme GAME_EVENT and after that there is no need to use input.txt file

Reply:
It's done.

Reply:
Let's delete the input.txt and schema.json files then